### PR TITLE
Audiounit from c to cpp

### PR DESCRIPTION
**Pull request by achronop**
Changes to change audiounit compiler from C to C++

### CODE REVIEW

File: src/cubeb_audiounit.cpp [hunk 1]
```diff
@@ -358,16 +358,18 @@ audiounit_output_callback(void * user_ptr,
...
 {
   cubeb * ctx;
   int r;
 
   *context = NULL;
 
-  ctx = calloc(1, sizeof(*ctx));
+  ctx = new cubeb;
   assert(ctx);
+  memset(ctx, 0, sizeof(cubeb));
```
Comment (padenot) [inline src/cubeb_audiounit.cpp]: Use `PodZero` from `cubeb_utils.h`.

File: src/cubeb_audiounit.cpp [hunk 2]
```diff
@@ -945,8 +948,9 @@ audiounit_stream_init(cubeb * context,
     }
   }
 
-  stm = calloc(1, sizeof(cubeb_stream));
+  stm = new cubeb_stream;
   assert(stm);
+  memset(stm, 0, sizeof(cubeb_stream));
```
Comment (padenot) [inline src/cubeb_audiounit.cpp]: `PodZero` here also.

File: src/cubeb_audiounit.cpp [hunk 3]
```diff
@@ -1419,10 +1423,11 @@ int audiounit_stream_get_current_device(cubeb_stream * stm,
     return CUBEB_ERROR;
   }
 
-  *device = malloc(sizeof(cubeb_device));
+  *device = new cubeb_device;
   if (!*device) {
     return CUBEB_ERROR;
   }
+  memset(*device, 0, sizeof(cubeb_device));
```
Comment (padenot) [inline src/cubeb_audiounit.cpp]: And here.

File: src/cubeb_ring_array.h [hunk 4]
```diff
@@ -64,7 +64,7 @@ ring_array_init(ring_array * ra,
   ra->tail = 0;
   ra->count = 0;
 
-  ra->buffer_array = calloc(ra->capacity, sizeof(AudioBuffer));
+  ra->buffer_array = (AudioBuffer *)calloc(ra->capacity, sizeof(AudioBuffer));
```
Comment (padenot) [inline src/cubeb_ring_array.h]: `new `+ `PodZero` please.

**Comment by padenot** [comment]
Can we use `new` and `PodZero` everywhere, as well as `static_cast` ?

File: src/cubeb_ring_array.h [hunk 5]
```diff
@@ -92,10 +92,10 @@ ring_array_destroy(ring_array * ra)
   }
   for (unsigned int i = 0; i < ra->capacity; ++i) {
     if (ra->buffer_array[i].mData) {
-      free(ra->buffer_array[i].mData);
+      operator delete(ra->buffer_array[i].mData);
```
Comment (padenot) [inline src/cubeb_ring_array.h]: Why do you need `operator` here ?

Comment (achronop) [inline src/cubeb_ring_array.h]: Because memory is allocated with `operator new`.

**Pull request merged**